src/gwt/src/org/rstudio/studio/client/workbench/views/vcs/git/dialog/GitReviewPresenter.java

```diff
@@ -520,6 +524,95 @@ public void onValueChange(ValueChangeEvent<Integer> event)
          @Override
          public void onClick(ClickEvent event)
          {
+            // record the current selected paths (as a set)
+            final Set<String> selectedPaths = new HashSet<String>();
+            selectedPaths.addAll(view_.getSelectedPaths());
+            
+            // first, double-check the file sizes of any files that are going
+            // to be added or modified (help the user avoid committing large
+            // files to the repository)
+            server_.gitFullStatus(new ServerRequestCallback<JsArray<StatusAndPathInfo>>()
+            {
+               @Override
+               public void onResponseReceived(JsArray<StatusAndPathInfo> response)
+               {
+                  // if this failed for some reason just go ahead with the commit
+                  if (response == null)
+                  {
+                     onCommit();
+                     return;
+                  }
+                  
+                  // iterate through the files and check for large modifications
+                  for (int i = 0, n = response.length(); i < n; i++)
+                  {
+                     StatusAndPathInfo info = response.get(i);
+                     
+                     // skip if this isn't a currently selected file for add
+                     if (!selectedPaths.contains(info.getPath()))
+                        continue;
+                     
+                     String status = info.getStatus();
+                     if (status.charAt(0) != 'A')
+                        continue;
+                     
+                     // warn if we're trying to commit a file >10MB in size
+                     double size = info.getFileSize();
+                     if (size > 10 * 1024 * 1024)
```

Comment:
Yes, it should be very straightforward! You just need to add a session option to the .conf file like this: https://github.com/rstudio/rstudio/blob/62b217a7115f98e933b4436a04cd5ac675a4f8c6/src/cpp/session/SessionOptions.cpp#L221-L223  ... and then read it into the SessionInfo object on client init: https://github.com/rstudio/rstudio/blob/62b217a7115f98e933b4436a04cd5ac675a4f8c6/src/cpp/session/SessionClientInit.cpp#L285